openage/convert/interface/visgrep.py

```diff
@@ -0,0 +1,389 @@
+                                     formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    PARSER.add_argument("-x", dest="off_x", default=0, metavar="X_OFF", type=int,
+                        help="Set x offset for detection matching")
+    PARSER.add_argument("-y", dest="off_y", default=0, metavar="Y_OFF", type=int,
+                        help="Set y offset for detection matching")
+    PARSER.add_argument("-I", dest="src_width", default=0, metavar="WIDTH", type=int,
+                        help="Set width for cropping the source image")
+    PARSER.add_argument("-J", dest="src_height", default=0, metavar="HEIGHT", type=int,
+                        help="Set height for cropping the source image")
+    PARSER.add_argument("-W", dest="sub_width", default=0, metavar="X_OFF_WIDTH", type=int,
+                        help="Set x offset width for detection matching")
+    PARSER.add_argument("-H", dest="sub_height", default=0, metavar="Y_OFF_HEIGHT", type=int,
+                        help="Set y offset height for detection matching")
+    PARSER.add_argument("-X", dest="start_x", default=0, metavar="START_X_OFF", type=int,
+                        help="Start scanning at X")
+    PARSER.add_argument("-Y", dest="start_y", default=0, metavar="START_Y_OFF", type=int,
+                        help="Start scanning at Y")
+    PARSER.add_argument("-u", dest="off_u", default=0, metavar="U_OFF", type=int,
+                        help="Set u offset for cropping patterns before use")
+    PARSER.add_argument("-v", dest="off_v", default=0, metavar="V_OFF", type=int,
+                        help="Set y offset for cropping patterns before use")
+    PARSER.add_argument("-M", dest="pattern_width", metavar="U_WIDTH", type=int,
+                        help="Set width of the cropped patterns")
+    PARSER.add_argument("-N", dest="pattern_height", metavar="V_HEIGHT", type=int,
+                        help="Set height of the cropped patterns")
+    PARSER.add_argument("-a", dest="scan_all", help="""Scan all patterns, not just after matching
+                        the detection pattern note: this method is much slower because we scan for
+                        all images at every pixel instead of just at detection points. Also, in
+                        this mode the detection image is ignored, there will be no matches
+                        for tile -1""", action="store_true")
+    PARSER.add_argument("-t", dest="tolerance", default=0, metavar="TOLERANCE", type=int,
+                        help="Set tolerance for 'fuzzy' matches, higher numbers are more tolerant")
+    PARSER.add_argument("-b", dest="show_badness",
+                        help="""Display 'badness' value, higher numbers mean match is less accurate,
+                        a badness value of 0 means the match is pixel-perfect""",
+                        action="store_true")
+    PARSER.add_argument("-d", metavar="DEBUGLEVEL", type=int, help="Print debug messages")
```

Comment:
Here you can use type=count, then `-dddd` will be level 4. I'd recommend using `-v` though except if you don't want to break the c-visgrep compat.